gateway-ha/src/main/java/io/trino/gateway/proxyserver/ProxyRequestHandler.java

```diff
@@ -171,46 +173,50 @@ private void performRequest(
 
         if (statementPaths.stream().anyMatch(request.getUri().getPath()::startsWith) && request.getMethod().equals(HttpMethod.POST)) {
             future = future.transform(response -> recordBackendForQueryId(request, response), executor);
+            if (includeClusterInfoInResponse) {
+                cookies.add(new NewCookie.Builder("trinoClusterHost").value(remoteUri.getHost()).build());
+            }
         }
 
         setupAsyncResponse(
                 asyncResponse,
-                future.transform(response -> buildResponse(response, oauth2GatewayCookie), executor)
+                future.transform(response -> buildResponse(response, cookies), executor)
                         .catching(ProxyException.class, e -> handleProxyException(request, e), directExecutor()));
     }
 
-    private ImmutableList<NewCookie> getOAuth2GatewayCookie(URI remoteUri, HttpServletRequest servletRequest)
+    private List<NewCookie> getOAuth2GatewayCookie(URI remoteUri, HttpServletRequest servletRequest)
     {
+        List<NewCookie> cookies = new ArrayList<>();
```

Comment:
Please use `ImmutableList.Builder` instead.

Reply:
Refer to my other comment.

Reply:
I'm suggesting `ImmutableList.Builder` not `ImmutableList`

Reply:
Actually, changes in this method aren't necessary. Please revert.

Reply:
Please suggest alternative instead.